intake_esm/core.py

```diff
@@ -1,195 +1,151 @@
-import inspect
+import fnmatch
+import logging
+import os
 import uuid
+from glob import glob
 
 import numpy as np
 import pandas as pd
 import xarray as xr
+import yaml
 from intake.catalog import Catalog
 from intake.catalog.local import LocalCatalogEntry
 from intake_xarray.netcdf import NetCDFSource
 
 from ._version import get_versions
-from .common import get_collection_def, get_subset, open_collection
-from .manage_collections import CESMCollections
+from .cesm import CESMCollection, CESMSource
+from .cmip import CMIPCollection, CMIPSource
+from .common import _get_built_collections, _open_collection
+from .config import INTAKE_ESM_CONFIG_FILE, SETTINGS
 
-__version__ = get_versions()["version"]
+__version__ = get_versions()['version']
 del get_versions
 
 
-class CesmMetadataStoreCatalog(Catalog):
-    """ CESM collection Metadata store """
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.DEBUG)
 
-    name = "cesm_metadatastore"
+
+class ESMMetadataStoreCatalog(Catalog):
+    """ESM collection Metadata store. This class servers as an entry point for `intake_esm`. """
+
+    name = 'esm_metadatastore'
     version = __version__
+    collection_types = {'cesm': CESMCollection, 'cmip': CMIPCollection}
+    SOURCES = {'cesm': 'intake_esm.cesm.CESMSource', 'cmip': 'intake_esm.esm.CMIPSource'}
```

Comment:
@andersy005: there is comparable `SOURCES` defined in config. Should it be in both places?

Reply:
Good catch! I will fix it